Zend/zend_exceptions.c

```diff
@@ -489,8 +490,8 @@ static void _build_trace_args(zval *arg, smart_str *str) /* {{{ */
 			break;
 		case IS_STRING:
 			smart_str_appendc(str, '\'');
-			smart_str_append_escaped(str, Z_STRVAL_P(arg), MIN(Z_STRLEN_P(arg), 15));
-			if (Z_STRLEN_P(arg) > 15) {
+			smart_str_append_escaped(str, Z_STRVAL_P(arg), MIN(Z_STRLEN_P(arg), PG(exception_string_param_max_len)));
+			if (Z_STRLEN_P(arg) > PG(exception_string_param_max_len)) {
```

Comment:
Should we avoid cutting the string it is within 3 chars of the limit? I don't think it makes sense to append `...` if the result will be longer than displaying the full string.

Reply:
No. For example, with exception_string_param_max_len == 0, a user is indicating not to display the contents of a string.

So `""` would be the empty string, and `"..."` would be something longer than the empty string.

And I'd want to keep existing stack traces identical, so =15 should show 15 characters from the original string.

Reply:
Fair enough.